dune/gdt/local/numerical-fluxes/vijayasundaram.hh

```diff
@@ -24,55 +24,95 @@ namespace Dune {
 namespace GDT {
 
 
-template <size_t d, size_t m = 1, class R = double>
-class NumericalVijayasundaramFlux : public NumericalFluxInterface<d, m, R>
+template <class I, size_t d, size_t m = 1, class R = double>
+class NumericalVijayasundaramFlux : public NumericalFluxInterface<I, d, m, R>
 {
-  using ThisType = NumericalVijayasundaramFlux<d, m, R>;
-  using BaseType = NumericalFluxInterface<d, m, R>;
+  using ThisType = NumericalVijayasundaramFlux;
+  using BaseType = NumericalFluxInterface<I, d, m, R>;
 
 public:
+  using typename BaseType::E;
   using typename BaseType::FluxType;
+  using typename BaseType::FunctionType;
+  using typename BaseType::LocalIntersectionCoords;
   using typename BaseType::PhysicalDomainType;
-  using typename BaseType::StateRangeType;
-
-  using FluxEigenDecompositionLambdaType = std::function<std::tuple<
+  using typename BaseType::StateType;
+
+  using FluxEigenDecompositionLambdaType =
+      std::function<std::tuple<std::vector<XT::Common::real_t<R>>,
+                               XT::Common::FieldMatrix<XT::Common::real_t<R>, m, m>,
+                               XT::Common::FieldMatrix<XT::Common::real_t<R>, m, m>>(
+          const FluxType&,
+          const E&,
+          const PhysicalDomainType&,
+          const FieldVector<R, m>&,
+          const FieldVector<double, d>&,
+          const XT::Common::Parameter& param)>;
+
+  using GlobalFluxEigenDecompositionLambdaType = std::function<std::tuple<
       std::vector<XT::Common::real_t<R>>,
       XT::Common::FieldMatrix<XT::Common::real_t<R>, m, m>,
       XT::Common::FieldMatrix<XT::Common::real_t<R>, m, m>>(
       const FluxType&, const FieldVector<R, m>&, const FieldVector<double, d>&, const XT::Common::Parameter& param)>;
 
-  NumericalVijayasundaramFlux(const FluxType& flx)
-    : BaseType(flx)
-    , flux_eigen_decomposition_([](const auto& f, const auto& w, const auto& n, const auto& param) {
+  static FluxEigenDecompositionLambdaType default_flux_eigen_decomposition()
+  {
+    return [](const FluxType& f,
+              const E& e,
```

Comment:
I will have to think a bit more about adjusting this numerical flux, since the `w` here is `f(0.5*(u + v))` which does not yet make sense for an x-dependent `f`. So for the moment I would throw in the ctor, if the `flux()` is `x_dependent()`, and let the rest remain as it was.

Reply:
Done.